Tests/ClientRuntimeTests/ClientRuntimeTests/MiddlewareTests/OperationStackTests.swift

```diff
@@ -31,58 +24,49 @@ class OperationStackTests: HttpRequestTestBase {
 
         var stack = OperationStack<MockInput, MockOutput, MockMiddlewareError>(id: "Test Operation")
         stack.initializeStep.intercept(position: .before, middleware: MockInitializeMiddleware(id: "TestInitializeMiddleware", callback: { _, _ in
-            currExpectCount = self.checkAndFulfill(currExpectCount, 1, expectation: expectInitializeMiddleware)
+            currExpectCount = self.checkAndFulfill(currExpectCount, 1)
```

Comment:
Debatable: We are still using expectation terminology without expectations. Maybe we can just replace it with two lines something like 
rename `currExpectCount` to `currCallbacksCount` and doing 

```suggestion
	  XCTAssert(currCallbackCount, 1)
	  currCallbackCount += 1
```
Or
renaming the function to something  like

`checkExpectedCountAndIncrement`

Reply:
I cleaned this up a bit, plus I added a check at the end (verifying the counter value) to prove that the callbacks were actually made.